File: sgi-webapp/src/app/core/models/csp/estado-solicitud.ts

import { marker } from '@biesbjerg/ngx-translate-extract-marker';
import { DateTime } from 'luxon';

export interface IEstadoSolicitud {
  /** Id */
  id: number;
  /** ID de la solicitud */
  solicitudId: number;
  /** Estado */
  estado: Estado;
  /** Fecha estado */
  fechaEstado: DateTime;
  /** Comentario */
  comentario: string;
}

export enum Estado {
  ADMITIDA_DEFINITIVA = 'ADMITIDA_DEFINITIVA',
  ADMITIDA_PROVISIONAL = 'ADMITIDA_PROVISIONAL',
  ALEGACION_FASE_ADMISION = 'ALEGACION_FASE_ADMISION',
  ALEGACION_FASE_PROVISIONAL = 'ALEGACION_FASE_PROVISIONAL',
  BORRADOR = 'BORRADOR',
  CONCEDIDA = 'CONCEDIDA',
  CONCEDIDA_PROVISIONAL = 'CONCEDIDA_PROVISIONAL',
  CONCEDIDA_PROVISIONAL_ALEGADA = 'CONCEDIDA_PROVISIONAL_ALEGADA',
  CONCEDIDA_PROVISIONAL_NO_ALEGADA = 'CONCEDIDA_PROVISIONAL_NO_ALEGADA',
  DENEGADA = 'DENEGADA',
  DENEGADA_PROVISIONAL = 'DENEGADA_PROVISIONAL',
  DENEGADA_PROVISIONAL_ALEGADA = 'DENEGADA_PROVISIONAL_ALEGADA',
  DENEGADA_PROVISIONAL_NO_ALEGADA = 'DENEGADA_PROVISIONAL_NO_ALEGADA',
  DESISTIDA = 'DESISTIDA',
  EXCLUIDA_DEFINITIVA = 'EXCLUIDA_DEFINITIVA',
  EXCLUIDA_PROVISIONAL = 'EXCLUIDA_PROVISIONAL',
  FIRMADA = 'FIRMADA',
  NEGOCIACION = 'NEGOCIACION',
  PRESENTADA_SUBSANACION = 'PRESENTADA_SUBSANACION',
  RECURSO_FASE_ADMISION = 'RECURSO_FASE_ADMISION',
  RECURSO_FASE_CONCESION = 'RECURSO_FASE_CONCESION',
  RENUNCIADA = 'RENUNCIADA',
  RESERVA = 'RESERVA',
  RESERVA_PROVISIONAL = 'RESERVA_PROVISIONAL',
  SOLICITADA = 'SOLICITADA',
  SUBSANACION = 'SUBSANACION',
  RECHAZADA = 'RECHAZADA',
  VALIDADA = 'VALIDADA'
}

export const ESTADO_MAP: Map<Estado, string> = new Map([
  [Estado.ADMITIDA_DEFINITIVA, marker(`csp.estado-solicitud.ADMITIDA_DEFINITIVA`)],
  [Estado.ADMITIDA_PROVISIONAL, marker(`csp.estado-solicitud.ADMITIDA_PROVISIONAL`)],
  [Estado.ALEGACION_FASE_ADMISION, marker(`csp.estado-solicitud.ALEGACION_FASE_ADMISION`)],
  [Estado.ALEGACION_FASE_PROVISIONAL, marker(`csp.estado-solicitud.ALEGACION_FASE_PROVISIONAL`)],
  [Estado.BORRADOR, marker(`csp.estado-solicitud.BORRADOR`)],
  [Estado.CONCEDIDA, marker(`csp.estado-solicitud.CONCEDIDA`)],
  [Estado.CONCEDIDA_PROVISIONAL, marker(`csp.estado-solicitud.CONCEDIDA_PROVISIONAL`)],
  [Estado.CONCEDIDA_PROVISIONAL_ALEGADA, marker(`csp.estado-solicitud.CONCEDIDA_PROVISIONAL_ALEGADA`)],
  [Estado.CONCEDIDA_PROVISIONAL_NO_ALEGADA, marker(`csp.estado-solicitud.CONCEDIDA_PROVISIONAL_NO_ALEGADA`)],
  [Estado.DENEGADA, marker(`csp.estado-solicitud.DENEGADA`)],
  [Estado.DENEGADA_PROVISIONAL, marker(`csp.estado-solicitud.DENEGADA_PROVISIONAL`)],
  [Estado.DENEGADA_PROVISIONAL_ALEGADA, marker(`csp.estado-solicitud.DENEGADA_PROVISIONAL_ALEGADA`)],
  [Estado.DENEGADA_PROVISIONAL_NO_ALEGADA, marker(`csp.estado-solicitud.DENEGADA_PROVISIONAL_NO_ALEGADA`)],
  [Estado.DESISTIDA, marker(`csp.estado-solicitud.DESISTIDA`)],
  [Estado.EXCLUIDA_DEFINITIVA, marker(`csp.estado-solicitud.EXCLUIDA_DEFINITIVA`)],
  [Estado.EXCLUIDA_PROVISIONAL, marker(`csp.estado-solicitud.EXCLUIDA_PROVISIONAL`)],
  [Estado.FIRMADA, marker(`csp.estado-solicitud.FIRMADA`)],
  [Estado.NEGOCIACION, marker(`csp.estado-solicitud.NEGOCIACION`)],
  [Estado.PRESENTADA_SUBSANACION, marker(`csp.estado-solicitud.PRESENTADA_SUBSANACION`)],
  [Estado.RECURSO_FASE_ADMISION, marker(`csp.estado-solicitud.RECURSO_FASE_ADMISION`)],
  [Estado.RECURSO_FASE_CONCESION, marker(`csp.estado-solicitud.RECURSO_FASE_CONCESION`)],
  [Estado.RENUNCIADA, marker(`csp.estado-solicitud.RENUNCIADA`)],
  [Estado.RESERVA, marker(`csp.estado-solicitud.RESERVA`)],
  [Estado.RESERVA_PROVISIONAL, marker(`csp.estado-solicitud.RESERVA_PROVISIONAL`)],
  [Estado.SOLICITADA, marker(`csp.estado-solicitud.SOLICITADA`)],
  [Estado.SUBSANACION, marker(`csp.estado-solicitud.SUBSANACION`)],
  [Estado.RECHAZADA, marker(`csp.estado-solicitud.RECHAZADA`)],
  [Estado.VALIDADA, marker(`csp.estado-solicitud.VALIDADA`)],
]);

File: sgi-csp-service/src/main/java/org/crue/hercules/sgi/csp/model/EstadoSolicitud.java
package org.crue.hercules.sgi.csp.model;

import java.time.Instant;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "estado_solicitud")
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EstadoSolicitud extends BaseEntity {

  /**
   * Serial version
   */
  private static final long serialVersionUID = 1L;

  /**
   * Enumerado tipo estado de las solicitudes.
   *
   */
  public enum Estado {
    /** Borrador */
    BORRADOR,
    /** Solicitrada */
    SOLICITADA,
    /** En subsanación */
    SUBSANACION,
    /** Presentada subsanación */
    PRESENTADA_SUBSANACION,
    /** Excluida provisional */
    EXCLUIDA_PROVISIONAL,
    /** Admitida provisional */
    ADMITIDA_PROVISIONAL,
    /** Alegación fase admisión */
    ALEGACION_FASE_ADMISION,
    /** Admitida definitiva */
    ADMITIDA_DEFINITIVA,
    /** DExcluida definitiva */
    EXCLUIDA_DEFINITIVA,
    /** Recurso fase admisión */
    RECURSO_FASE_ADMISION,
    /** Concedida provisional */
    CONCEDIDA_PROVISIONAL,
    /** Denegada provisional */
    DENEGADA_PROVISIONAL,
    /** Alegación fase provisional */
    ALEGACION_FASE_PROVISIONAL,
    /** Concedida provisional alegada */
    CONCEDIDA_PROVISIONAL_ALEGADA,
    /** Concedida provisional no alegada */
    CONCEDIDA_PROVISIONAL_NO_ALEGADA,
    /** Denegada provisional alegada */
    DENEGADA_PROVISIONAL_ALEGADA,
    /** Denegada provisional no alegada */
    DENEGADA_PROVISIONAL_NO_ALEGADA,
    /** Desistida */
    DESISTIDA,
    /** Reserva provisional */
    RESERVA_PROVISIONAL,
    /** En negociación */
    NEGOCIACION,
    /** Concedida */
    CONCEDIDA,
    /** Denegada */
    DENEGADA,
    /** Recurso fase concesión */
    RECURSO_FASE_CONCESION,
    /** Reserva */
    RESERVA,
    /** Firmada */
    FIRMADA,
    /** Renunciada */
    RENUNCIADA,
    /** Rechazada */
    RECHAZADA,
    /** Validada */
    VALIDADA;
  }

  /** Id */
  @Id
  @Column(name = "id", nullable = false)
  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "estado_solicitud_seq")
  @SequenceGenerator(name = "estado_solicitud_seq", sequenceName = "estado_solicitud_seq", allocationSize = 1)
  private Long id;

  /** Solicitud Id */
  @Column(name = "solicitud_id", nullable = false)
  private Long solicitudId;

  /** Tipo estado solicitud */
  @Column(name = "estado", length = 50, nullable = false)
  @Enumerated(EnumType.STRING)
  @NotNull
  private Estado estado;

  /** Fecha. */
  @Column(name = "fecha_estado", nullable = false)
  private Instant fechaEstado;

  /** Comentario */
  @Column(name = "comentario", length = 2000, nullable = true)
  @Size(max = 2000)
  private String comentario;

  // Relation mappings for JPA metamodel generation only
  @ManyToOne
  @JoinColumn(name = "solicitud_id", insertable = false, updatable = false, foreignKey = @ForeignKey(name = "FK_ESTADOSOLICITUD_SOLICITUD"))
  @Getter(AccessLevel.NONE)
  @Setter(AccessLevel.NONE)
  private final Solicitud solicitud = null;
}